pkg/storage/mvcc_value.go

```diff
@@ -216,6 +237,50 @@ func EncodeMVCCValue(v MVCCValue) ([]byte, error) {
 	return buf, nil
 }
 
+// EncodeMVCCValueToBuf encodes an MVCCValue into its Pebble
+// representation. See the comment on MVCCValue for a description of
+// the encoding scheme.
+//
+// If extended encoding is required, the given buffer will be used. If
+// the provided buffer is not large enough an error is
```

Comment:
2¢:
Instead of an error I'd just allocate a bigger buffer and return that. This makes your TODO easily implemented by  `EncodeMVCCValueToBuf(x) { return EncodeMVCCValueToBuf(x, nil) }`.

And then I'd probably swap the bufalloc for just `scratch []byte` and call it as `scratch, err =  EncodeMVCCValueToBuf(x, scratch[:0])` so that if it allocates, you keep the result?

/2¢

Reply:
Very nice. This will clean things up a lot.